components/brave_wallet/browser/cardano/cardano_get_utxos_task.cc

```diff
@@ -82,17 +73,18 @@ void GetCardanoUtxosTask::OnGetUtxoList(
 }
 
 void GetCardanoUtxosTask::WorkOnTask() {
-  if (error_) {
-    std::move(callback_).Run(this, base::unexpected(std::move(*error_)));
-    return;
-  }
-
   if (result_) {
-    std::move(callback_).Run(this, base::ok(std::move(*result_)));
+    std::move(callback_).Run(base::ok(std::move(*result_)));
     return;
   }
 
   MaybeSendRequests();
 }
 
+void GetCardanoUtxosTask::StopWithError(std::string error_string) {
+  base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
+      FROM_HERE, base::BindOnce(std::move(callback_),
+                                base::unexpected(std::move(error_string))));
+}
```

Comment:
can we do `base::BindPostTaskToCurrentDefault` when setting callback_? same with `CreateCardanoTransactionTask`